src/TensorOneD.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2015 - 2022, Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#include <math.h>
+#include <algorithm>
+#include <functional>
+#include <numeric>
+#include "geopm/Exception.hpp"
+
+#include "TensorOneD.hpp"
+
+namespace geopm
+{
+    TensorOneD::TensorOneD()
+    {
+    }
+
+    TensorOneD::TensorOneD(int dim)
+    {
+        set_dim(dim);
+    }
+
+    TensorOneD::TensorOneD(const TensorOneD &other)
+    {
+        m_vec = other.m_vec;
+    }
+
+    TensorOneD::TensorOneD(json11::Json input)
+    {
+        set_dim(input.array_items().size());
+        for (int idx = 0; idx < (int)m_vec.size(); idx++) {
```

Comment:
Earlier recommendation about avoiding C-style casts applies here as well. Declare ``idx`` as the vector's ``size_type``, or use ``size_t``.